app/notification.go

```diff
@@ -36,6 +36,22 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 		close(cmnchan)
 	}()
 
+	var gchan chan store.StoreResult
+	if a.allowGroupMentions(post) {
```

Comment:
If the user does not have access to group mentions no extra query will be made for groups and the groups map will just be empty